connection.cpp

```diff
@@ -157,12 +161,7 @@ void Connection::sync(int timeout)
     auto job = d->syncJob =
             callApi<SyncJob>(d->data->lastEvent(), filter, timeout);
     connect( job, &SyncJob::success, [=] () {
-        d->data->setLastEvent(job->nextBatch());
-        for( auto&& roomData: job->takeRoomData() )
-        {
-            if ( auto* r = provideRoom(roomData.roomId) )
-                r->updateData(std::move(roomData));
-        }
+        onSyncSuccess(*job->data());
```

Comment:
And, as mentioned in the Matrix room, `SyncJob::d` member need not be a pointer anymore, since the underlying structure is defined in the header file now and it has the same lifetime. `SyncData d;` will work equally well, and the weirdness of `std::move(*d);` will not be there.